### ISSUE TITLE
Add new component specification: Select

### ISSUE THREAD

**Issue by chrisdholt** [microsoft/fast#3751]
A new specification needs to be created for "Select" as a foundational component.

This issue should be closed with the introduction of the Select specification, and a new issue should be opened to create the control once the spec is complete.

**Comment by MarcSkovMadsen** [comment]
Listening in. I need this component for a specific application I am working on.

**Comment by MarcSkovMadsen** [comment]
I need the `fast-select` :-)

![image](https://user-images.githubusercontent.com/42288570/92991545-ddda1800-f4e4-11ea-8d40-ebf80527dc02.png)

**Comment by EisenbergEffect** [comment]
@radium-v Can we close this one as done?

**Comment by radium-v** [comment]
@EisenbergEffect The `fast-select` and `fast-option` components are complete but the spec side still needs to be addressed. #3619 is an older PR for the spec side.

**Comment by EisenbergEffect** [comment]
@radium-v Any notion of how much work is left on the spec? Is it something that could get finished this week or should we push it out to the next milestone?

**Comment by radium-v** [comment]
@EisenbergEffect I'll pick up the spec now and update it to be in sync with the select component.

**Comment by EisenbergEffect** [comment]
Thanks!
